core/vm/contracts.go

```diff
@@ -358,3 +374,68 @@ func (c *bn256Pairing) Run(input []byte) ([]byte, error) {
 	}
 	return false32Byte, nil
 }
+
+type blake2F struct{}
+
+func (c *blake2F) RequiredGas(input []byte) uint64 {
+	if len(input) != blake2FInputLength {
+		// Input is malformed, we can't read the number of rounds.
+		// Precompile can't be executed so we set its price to 0.
+		return 0
+	}
+
+	rounds := binary.BigEndian.Uint32(input[209:213])
+	return uint64(rounds)
+}
+
+var (
+	blake2FInputLength = 213
+
+	errBlake2FIncorrectInputLength = errors.New(
+		"input length for Blake2 F precompile should be exactly 213 bytes",
+	)
+)
+
+func (c *blake2F) Run(input []byte) ([]byte, error) {
+	if len(input) != blake2FInputLength {
+		return nil, errBlake2FIncorrectInputLength
+	}
+
+	var h [8]uint64
+	h[0] = binary.BigEndian.Uint64(input[0:8])
+	h[1] = binary.BigEndian.Uint64(input[8:16])
+	h[2] = binary.BigEndian.Uint64(input[16:24])
+	h[3] = binary.BigEndian.Uint64(input[24:32])
+	h[4] = binary.BigEndian.Uint64(input[32:40])
+	h[5] = binary.BigEndian.Uint64(input[40:48])
+	h[6] = binary.BigEndian.Uint64(input[48:56])
+	h[7] = binary.BigEndian.Uint64(input[56:64])
+
+	var m [blake2.BlockSize]byte
+	copy(m[:], input[64:192])
+
+	var t [2]uint64
+	t[0] = binary.BigEndian.Uint64(input[192:200])
+	t[1] = binary.BigEndian.Uint64(input[200:208])
+
+	var f bool
+	if input[208] == 0x00000001 {
```

Comment:
```suggestion
      if input[208] != 0x00000000 {
```

Reply:
This will be checked in a different way, just `!= 0 `